src/mame/audio/vicdual-97271p.cpp

```diff
@@ -0,0 +1,206 @@
+// license:BSD-3-Clause
+// copyright-holders:Ariane Fugmann
+
+/*
+N-Sub Oscillator 97271-P
+-----------------------------
+for the time being this is a simple sample player based on the work by MASH.
+*/
+
+#include "emu.h"
+#include "speaker.h"
+#include "audio/vicdual-97271p.h"
+
+#define S97271P_TAG     "s97271p"
+
+//**************************************************************************
+//  GLOBAL VARIABLES
+//**************************************************************************
+
+const device_type S97271P = device_creator<s97271p_device>;
+
+/* bit definitions - sound effect drive outputs */
+#define S97271P_WARNING      0x01
+#define S97271P_SONAR        0x02
+#define S97271P_LAUNCH       0x04
+#define S97271P_EXPL_L       0x08
+#define S97271P_EXPL_S       0x10
+#define S97271P_BONUS        0x20
+#define S97271P_CODE         0x40
+#define S97271P_BOAT         0x80
+
+/* sample file names */
+static const char *const nsub_sample_names[] =
+{
+	"*nsub",
+	"SND_EXPL_L0",
+	"SND_EXPL_L1",
+	"SND_SONAR",
+	"SND_LAUNCH0",
+	"SND_LAUNCH1",
+	"SND_WARNING0",
+	"SND_WARNING1",
+	"SND_EXPL_S0",
+	"SND_EXPL_S1",
+	"SND_BONUS0",
+	"SND_BONUS1",
+	"SND_CODE",
+	"SND_BOAT",
+	nullptr
+};
+
+/* sample ids - must match sample file name table above */
+enum 
+{
+	SND_EXPL_L0 = 0,
+	SND_EXPL_L1,
+	SND_SONAR,
+	SND_LAUNCH0,
+	SND_LAUNCH1,
+	SND_WARNING0,
+	SND_WARNING1,
+	SND_EXPL_S0,
+	SND_EXPL_S1,
+	SND_BONUS0,
+	SND_BONUS1,
+	SND_CODE,
+	SND_BOAT
+};
+
+
+//**************************************************************************
+//  MACHINE FRAGMENTS
+//**************************************************************************
+
+MACHINE_CONFIG_FRAGMENT( nsub_audio )
```

Comment:
This should be `static` or inside an anonymous namespace.  There's no need for it to be exported.